docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<img src = "images/ArchitectureDiagram.png">
 
+The above **Architecture Diagram** provides the high-level design of the application.
+
+### Main components
+
+The application comprises the following components:
+
+* `Main`, which handles program startup and shutdown, and also interactions between other components.
+* `UI`, which handles user input and showing messages to the user.
+* `Storage`, which handles the loading and saving of data upon program startup and shutdown.
+* `Parser`, which converts user input into commands.
+* `Command`, which are executed to modify the data stored in the program.
+* `Event`, which stores the program's data.
+
+### Interactions between components
+
+<img src = "images/ArchitectureSequenceDiagram.png">
+
+The above **Sequence Diagram** shows how the different components of the system interact with one
+another in the scenario when the command `add -e event -t 1200 -v venue` is executed.
```

Comment:
For your sequence diagram here, to better align yourselves with the notation suggested in the textbook, might it be better to add colons to indicate that these are actual instances? For example, `:Parser`. From the textbook as well, I noticed that there was no need to specify the return type of the object that is being returned from a method.

Might you consider adding a `User` participant that interacts with `Ui` instead of `Main`?

Reply:
I agree with adding colons to indicate that these are actual instances/objects instead of classes. This comment applies to all other sequence diagrams throughout the DG too.